pkg/controllers/workgenerator/controller_test.go

```diff
@@ -143,6 +147,199 @@ func TestGetWorkNamePrefixFromSnapshotName(t *testing.T) {
+		cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
+		cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ManagedFields"),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.IgnoreFields(fleetv1beta1.WorkloadTemplate{}, "Manifests"),
+	}
+
+	testDeployment := appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       utils.DeploymentKind,
+			APIVersion: utils.DeploymentGVK.GroupVersion().String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testDeployment",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:        ptr.To(int32(2)),
+			MinReadySeconds: 5,
+		},
+	}
+	newWork := &fleetv1beta1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				fleetv1beta1.ParentResourceSnapshotIndexLabel: "1",
+			},
+			Annotations: map[string]string{
+				fleetv1beta1.ParentResourceSnapshotNameAnnotation:                "snapshot-1",
+				fleetv1beta1.ParentClusterResourceOverrideSnapshotHashAnnotation: "hash1",
+				fleetv1beta1.ParentResourceOverrideSnapshotHashAnnotation:        "hash2",
+			},
+		},
+		Spec: fleetv1beta1.WorkSpec{
+			Workload: fleetv1beta1.WorkloadTemplate{
+				Manifests: []fleetv1beta1.Manifest{{RawExtension: runtime.RawExtension{Object: &testDeployment}}},
+			},
+		},
+	}
+
+	resourceSnapshot := &fleetv1beta1.ClusterResourceSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "snapshot-1",
+			Labels: map[string]string{
+				fleetv1beta1.ResourceIndexLabel: "1",
+			},
+		},
+	}
+
+	tests := []struct {
+		name          string
+		existingWork  *fleetv1beta1.Work
+		expectChanged bool
+	}{
+		{
+			name:          "Create new work when existing work is nil",
+			existingWork:  nil,
+			expectChanged: true,
+		},
+		{
+			name: "Update existing work with new annotations",
+			existingWork: &fleetv1beta1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      workName,
+					Namespace: namespace,
+					Labels: map[string]string{
+						fleetv1beta1.ParentResourceSnapshotIndexLabel: "1",
+					},
+				},
+				Spec: fleetv1beta1.WorkSpec{
+					Workload: fleetv1beta1.WorkloadTemplate{
+						Manifests: []fleetv1beta1.Manifest{{RawExtension: runtime.RawExtension{Raw: []byte("{}")}}},
+					},
+				},
+			},
+			expectChanged: true,
+		},
+		{
+			name: "Update existing work even if it does not have the resource snapshot label",
+			existingWork: &fleetv1beta1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      workName,
+					Namespace: namespace,
+					Annotations: map[string]string{
+						fleetv1beta1.ParentClusterResourceOverrideSnapshotHashAnnotation: "hash1",
+					},
+				},
+				Spec: fleetv1beta1.WorkSpec{
+					Workload: fleetv1beta1.WorkloadTemplate{
+						Manifests: []fleetv1beta1.Manifest{{RawExtension: runtime.RawExtension{Raw: []byte("{}")}}},
+					},
+				},
+			},
+			expectChanged: true,
+		},
+
+		{
+			name: "Update existing work even if it does not have the resource snapshot label",
```

Comment:
Hi Ryan! This case is a duplicate of that in line 235 I think; is it supposed to be about annotations?